src/ovl0/ovl0_2_5_2.c

```diff
@@ -627,126 +608,106 @@ void func_80008830(struct Animation *anim, struct AnimStack *element) {
     anim->unk6C = element;
 }
 
-#ifdef MIPS_TO_C
-//generated by mips_to_c commit e0e006e8858ba357d1dcb4dc64f038b7df278aa6
-void func_80008840(void *arg0, void *arg1) {
-    *arg1 = (void *) arg0->unk90;
-    arg0->unk90 = arg1;
+// TODO: is this an existing struct instead of a brand new one?
+struct unk80008840 {
+    u8 padding[0x90];
+    struct AnimStack *unk90;
+};
+
+void func_80008840(struct unk80008840 *arg0, struct AnimStack *element) {

```

Comment:
My theory on this function and `func_80008850` is that they handle whatever the game calls `MObj` and `DObj`, rather than AnimStack/`Aobj`

Reply:
Following the code paths shows that func_80008840 is called by func_800098AC, which passes it's own arg0 as the first argument and the return value of func_800087D8 as the second argument. func_800087D8 returns AObj as per the error text, so the function in question's second argument definitely is AObj, so I'll switch that.
func_800087D8 (the one that calls the function in question) is itself called by func_8000D35C, which seems to pass in it's own first argument as the first argument to func_800087D8. From there I wasn't able to get a clear idea of the struct being used, so I think using that temporarily unknown struct as the first argument is fine.

Reply:
Interesting, after another look there's 2 sets of AObj/AnimStack functions

I wonder what's being handled in the struct with an unk90 field then, since Animation is already something defined (maybe MObj handles matrix transformations for animations?) 🤔